### PR TITLE
enable tidb config by default

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
Fix issue #228 

### What is changed and how it works?
Enable integrated TiDB experimental config by default 

### Check List 

Tests 

 - Manual test 
Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: pkg/task/restore.go [hunk 1]
```diff
@@ -467,3 +462,18 @@ func RunRestoreTiflashReplica(c context.Context, g glue.Glue, cmdName string, cf
...
 	return nil
 }
+
+func enableTiDBConfig() {
+	// set max-index-length before execute DDLs and create tables
+	// we set this value to max(3072*4), otherwise we might not restore table
+	// when upstream and downstream both set this value greater than default(3072)
+	conf := config.GetGlobalConfig()
+	conf.MaxIndexLength = config.DefMaxOfMaxIndexLength
+	log.Warn("set max-index-length to max(3072*4) to skip check index length in DDL")
```
Comment (kennytm) [inline pkg/task/restore.go:472]: Perhaps this warning should also mention the experimental features. But I'm not sure about the point of having such warning at all when it's always emitted @5kbpers.

**Comment by 3pointer** [comment]
> Add a test case for #228. Rest LGTM.

okay, I'm working on it

**Comment by kennytm** [comment]
One more thing: can you also enable `alter-primary-key = true`? Without this setting the alter-pk related DDLs cannot be executed during incremental restore 😂.

**Comment by 3pointer** [comment]
> One more thing: can you also enable `alter-primary-key = true`? Without this setting the alter-pk related DDLs cannot be executed during incremental restore 😂.

okay, I'll enable it and add related test

**Comment by 3pointer** [comment]
> LGTM.
> 
> But I'm a bit worried that DDLs are executed before the CREATE TABLEs, meaning that in a _non_-alter-pk server if the incremental backup contained a CREATE TABLE the rows will be wrong.

we should consider to get restore cluster config to fix it, should I file an issue?

**Comment by kennytm** [comment]
Yes that would be the ideal case.

Alternatively we could special case all CREATE TABLE DDLs, such that if a table is successfully constructed, we will DROP that table before calling CreateDatabase(). 

Let's file an issue, and if this breaks non-alter-pk CREATE TABLE probably don't enable alter-pk in BR yet 😢

**Comment by 3pointer** [comment]
> Yes that would be the ideal case.
> 
> Alternatively we could special case all CREATE TABLE DDLs, such that if a table is successfully constructed, we will DROP that table before calling CreateDatabase().
> 
> Let's file an issue, and if this breaks non-alter-pk CREATE TABLE probably don't enable alter-pk in BR yet 😢

Even we can get restore cluster config `alter-pk`, we still can't simply enable alter-pk or not. because this config may change at created time and we didn't exactly know the table format. we need to get table format and save it into `backupmeta`, so I think I need file a tidb issue, to provide this format for BR

**Comment by kennytm** [comment]
There is a TableInfo field in the DDL Job which we can use in the next PR.

**Pull request merged**